tests/conftest.py

```diff
@@ -5,7 +5,891 @@
+    ExternalIntegration,
+    ExternalIntegrationLink
+)
+from ..model.constants import MediaTypes
+from ..model.contributor import Contributor
+from ..model.coverage import (
+    CoverageRecord,
+    Timestamp,
+    WorkCoverageRecord
+)
+from ..model.customlist import CustomList
+from ..model.datasource import DataSource
+from ..model.edition import Edition
+from ..model.identifier import Identifier
+from ..model.integrationclient import IntegrationClient
+from ..model.library import Library
+from ..model.licensing import (
+    DeliveryMechanism,
+    License,
+    LicensePool,
+    LicensePoolDeliveryMechanism,
+    RightsStatus
+)
+from ..model.patron import Patron
+from ..model.resource import (
+    Hyperlink,
+    Representation
+)
+from ..model.work import Work
+from ..util.datetime_helpers import utc_now
 
 # Pull in the session_fixture defined in core/testing.py
 # which does the database setup and initialization
 pytest_plugins = ["{}.testing".format(basename(dirname(dirname(abspath(__file__)))))]
+
+
+@pytest.fixture(autouse=True, scope="session")
+def init_test_db():
+    db_url = Configuration.database_url()
+    engine = create_engine(db_url)
+
+    for table in reversed(Base.metadata.sorted_tables):
+        try:
+            engine.execute(table.delete())
+        except ProgrammingError:
+            pass
+
+    with engine.connect() as conn:
+        Base.metadata.create_all(conn)
+
+    # Avoid CannotLoadConfiguration errors related to CDN integrations.
+    Configuration.instance[Configuration.INTEGRATIONS] = Configuration.instance.get(
+        Configuration.INTEGRATIONS, {}
+    )
+    Configuration.instance[Configuration.INTEGRATIONS][ExternalIntegration.CDN] = {}
+    Configuration.instance[Configuration.CDNS_LOADED_FROM_DATABASE] = True
+
+    engine.dispose()
+
+
+@pytest.fixture
+def db_engine():
+    db_url = Configuration.database_url()
+    engine = create_engine(db_url)
+    yield engine
+    engine.dispose()
+
+
+@pytest.fixture
+def db_session(db_engine):
+    with db_engine.connect() as connection:
+        transaction = connection.begin_nested()
+        session = Session(connection)
+
+        yield session
+
+        transaction.rollback()
+
+        Collection.reset_cache()
+        ConfigurationSetting.reset_cache()
+        DataSource.reset_cache()
+        DeliveryMechanism.reset_cache()
+        ExternalIntegration.reset_cache()
+        Genre.reset_cache()
+        Library.reset_cache()
+
+        for key in [
+                Configuration.SITE_CONFIGURATION_LAST_UPDATE,
+                Configuration.LAST_CHECKED_FOR_SITE_CONFIGURATION_UPDATE
+        ]:
+            if key in Configuration.instance:
+                del(Configuration.instance[key])
+
+        session.close()
+
+
+@pytest.fixture
+def initalize_data(db_session):
+    # Should all of core/model/__init__.py::SessionManager::initialize_data
+    # be in here instead of splitting out DeliveryMechanisms from DataSource & Genres?
```

Comment:
I think it should. The built-in DataSources and Genres should be in the database from the beginning and any test should be able to rely on their presence. Creating them on a test-by-test basis (which I assume is what happens now) is also a drag on performance.

Reply:
Same for the delivery mechanisms. I'm OK with leaving the other fixtures in place and making them inputs into this one.

Although, since this fixture takes db_session as an input, does that mean these objects are being created inside the database session and they get rolled back when the test is over? It would be faster and conceptually preferable to create these "built-in" database objects with Pytest session scope so they are part of the database that gets reverted to after a test runs.